### PR TITLE
[osx install script] fix stop/start race

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Fix a bug where the agent would fail to restart at the end of the installation of the osx script.

https://github.com/DataDog/datadog-agent/blob/e6163b0afc09a0dee6425aa79b8fd6220f6c898c/cmd/agent/install_mac_os.sh#L105-L106

`launchctl stop` is non-blocking and running `launchctl start` immediately after will do nothing.

This lead to a race between the shutdown of the agent and the auto-restart of the agent systray app : https://github.com/DataDog/datadog-agent/blob/e6163b0afc09a0dee6425aa79b8fd6220f6c898c/cmd/agent/gui/systray/Sources/gui.swift#L103

This PR adds some logic to wait until the agent is fully stop before attempting to start it again.

### CODE REVIEW

File: cmd/agent/install_mac_os.sh [hunk 1]
```diff
@@ -103,6 +103,16 @@ if egrep 'api_key:( APIKEY)?$' "/opt/datadog-agent/etc/datadog.yaml" > /dev/null
...
+
+    # Wait for the agent to fully stop
+    retry=0
+    until [ $retry -ge 5 ]
+    do
+        curl -m 5 -o /dev/null -s -I http://127.0.0.1:5002 || break
+        retry=$[$retry+1]
+        sleep 5
+    done
+
```
Comment (olivielpeau) [inline cmd/agent/install_mac_os.sh]: could we print something if the agent still wasn't stopped after all the retries?

**Review by olivielpeau — COMMENTED** [review]
LGTM, just one comment

File: cmd/agent/install_mac_os.sh [hunk 2]
```diff
@@ -106,12 +106,16 @@ if egrep 'api_key:( APIKEY)?$' "/opt/datadog-agent/etc/datadog.yaml" > /dev/null
...
-    until [ $retry -ge 5 ]
-    do
+    until [ $retry -ge 5 ]; do
         curl -m 5 -o /dev/null -s -I http://127.0.0.1:5002 || break
         retry=$[$retry+1]
         sleep 5
     done
+    if [ $retry -ge 5 ]; then
+        printf "\n\033[33mCould not restart the agent.
+You may have to restart it manualy using the systray app or the \
```
Comment (olivielpeau) [inline cmd/agent/install_mac_os.sh]: typo: `manually`.

Also, the `\` at the end of the line is probably unintentional

**Review by olivielpeau — COMMENTED** [review]
typos :p

**Review by ofek — APPROVED** [review]
Thanks!

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**